code/game/mecha/working/clarke.dm

```diff
@@ -48,6 +48,24 @@
 		var/mob/living/brain/B = M.brainmob
 		hud.add_hud_to(B)
 
+/obj/mecha/working/clarke/play_stepsound(direction)
+	if(prob(30))
+		floor_damage()
+	. = ..()
+
+/obj/mecha/working/clarke/mechturn(direction)
+	if(prob(80))
+		floor_damage()
+	. = ..()
+
+/obj/mecha/working/clarke/proc/floor_damage()
+	var/turf/open/floor/tile = get_turf(src)
+	if(!istype(tile, /turf/open/floor))
+		return
+	if(tile.icon_plating == "basalt") //Find me a better way to tell a tile is basalt, and I'll use it
```

Comment:
`istype(tile, /turf/open/floor/plating/asteroid/basalt)` ?

Reply:
better use `asteroid`.
and yes, best way, to add, `turf/proc/damage_by_track(obj/mecha/tracker)` and all all behavior to turfs that can be damaged by tracks, and call `tile.damage_by_track(src)` here.

Reply:
Added a new proc to /turf/open/floor called crush(). It just calls break_tile(), but gets overridden by asteroid tiles. Clarke no longer checks for asteroid tiles, and just calls crush() and lets the floor tile figure it out.